File: TeamCubing.Domain/Models/RoomSettings.cs
using Newtonsoft.Json;

namespace TeamCubing.Domain.Models;

public class RoomSettings
{
    [JsonProperty("puzzle")]
    public RoomPuzzle Puzzle { get; set; }

    [JsonProperty("isOpen")]
    public bool IsOpen { get; set; }

    [JsonProperty("enableSolveTimeLimit")]
    public bool EnableSolveTimeLimit { get; set; }

    [JsonProperty("usersLimit")]
    public int UsersLimit { get; set; }
}

File: src/TeamCubing.API/ClientApp/src/app/models/roomSettings.ts
export interface RoomSettings {
  puzzle: RoomPuzzle;
  isOpen: boolean;
  enableSolveTimeLimit: boolean;
  usersLimit: number;
}

export enum RoomPuzzle {
  ThreeByThreeCube = 3,
  TwoByTwoCube = 2,
  FourByFourCube = 4,
  FiveByFiveCube = 5,
  SixBySixCube = 6,
  SevenBySevenCube = 7,
  Megaminx = 12,
}
